### DISCUSSION TITLE
XelPlayer reloaded

### DISCUSSION THREAD

**Discussion by rchastain** [mse-org/mseide-msegui#72] [Show and tell]
I uploaded [here](https://codeberg.org/rchastain/xelplayer) the source code of XelPlayer, a crossplatform GUI for MPlayer, by Alexandre Minoshi.

It compiles, but I am not sure that it works. It will be a good exercise to update it.

Help is welcome.

**Comment by rchastain** [comment]
First problem. I cannot open almost any form. For example, if I open _main.pas,_ and try to toggle from unit to form, I get this message:

![classtype-tmplayer-not-found](https://github.com/mse-org/mseide-msegui/assets/13957204/3f6a6638-c814-47c3-bc06-e4d22e775107)

I need time to understand what happens. If you have ideas...

**Comment by rchastain** [comment]
I see that **tmplayer** is defined in _alsmplayer.pas_.

```pascal
// alsmplayer.pas
type
  tmplayer = class(tcomponent)
```

Maybe is it necessary to build an IDE with that component? I don't know how to do that. I will search.

**Comment by fredvs** [comment]
Re-hello Roland.

To install new component, you may follow the tips of Leandro:
https://github.com/bgrabitmap/bgragui#mseide-instructions

**Reply by rchastain** [reply]
Interesting. Thanks for the link.

**Comment by rchastain** [comment]
I made a copy of **mymseide** project, named it **xelide**, replaced _mybutton.pas_ with _alsmplayer.pas_ and... it works!

![main-mfm](https://github.com/mse-org/mseide-msegui/assets/13957204/1a624c4a-c650-4302-87ae-36fa902894fd)

**Reply by fredvs** [reply]
Excellent!

**Comment by fredvs** [comment]
Hello Roland.

Here, ignoring the class not found when trying to switch unit<>from and fixing the "uses" in unit mseucs2toru.pas with this

```
uses
 classes,mclasses,msetypes{$ifdef FPC},strings{$endif},typinfo,sysutils,
 msestrings;
```

it compiles and runs (but all is in Russian).

But it runs only once, at next try the program is not loaded, maybe a protection somewhere in code.

**Comment by fredvs** [comment]
Hum, after rebooting, it runs at each try now.
But there is surely some things to adapt (and understand what it says in Russian).
I let you discover the perl... ;-)

![xelplayer](https://github.com/mse-org/mseide-msegui/assets/3421249/7b3a64d5-a9b1-424f-ba4f-620bcb89f4d8)

**Comment by fredvs** [comment]
Hola Roland!

Tip: MSEgui has a big familly now so maybe in code you may change all things like this:
```
{$IFDEF linux}
  fmplayer := '/usr/bin/mplayer';
{$ENDIF}
```

with this:

```
{$IFDEF unix}
if fileexists('/usr/bin/mplayer') then
  fmplayer := '/usr/bin/mplayer' else
if fileexists('/usr/local/bin/mplayer') then
  fmplayer := '/usr/local/bin/mplayer' else
  showmessage('mplayer is not installed!');
{$ENDIF}
```

;-)
